app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -80,6 +82,33 @@ class HomeFragment : Fragment() {
             }
             override fun onTransitionCompleted(p0: MotionLayout?, p1: Int) { }
         })
+
+        privateBrowsingButton.setOnClickListener {
+            // Get the current private mode preference and flip it
+            PreferenceManager.getDefaultSharedPreferences(context!!)
+                .edit()
+                .putBoolean(getString(R.string.pref_key_private_mode),
+                    !PreferenceManager.getDefaultSharedPreferences(context!!)
+                    .getBoolean(getString(R.string.pref_key_private_mode), false))
+                .apply()
+
+            applyPrivateModeIfNecessary()
+        }
+    }
+
+    private fun applyPrivateModeIfNecessary() {
+        val isPrivate = PreferenceManager.getDefaultSharedPreferences(context!!)
+            .getBoolean(getString(R.string.pref_key_private_mode), false)
+
+        if (isPrivate) {
+            PreferenceManager.getDefaultSharedPreferences(context)
+                .edit().putInt(activity!!.getString(R.string.pref_key_theme), R.style.PrivateModeTheme).apply()
+        } else {
+            PreferenceManager.getDefaultSharedPreferences(activity!!)
+                .edit().putInt(activity!!.getString(R.string.pref_key_theme), R.style.LightAppTheme).apply()
+        }
+
+        activity!!.recreate()
```

Comment:
Is there a way to apply re-apply a theme without recreating the entire activity? I can spend some time with you tomorrow brainstorming

Reply:
From the research I did, it looks like the only way to make sure *all aspects* of the theme are applied is a recreate. Happy to brainstorm and see if we can find a workaround 😄